### ISSUE TITLE
No rc2 build yet?

### ISSUE THREAD

**Issue by Gillardo** [xunit/xunit#736]
I have this on a project using the rc2 nightly builds.  Unfortunatly all my tests have stopped working (cant even be found) because xunit uses rc1 dependencies.  When will this be updated to use rc2 dependencies?

**Comment by Gillardo** [comment]
Its specifically the xunit.runner.dnx package i mean

**Comment by bradwilson** [comment]
Tried a CI build? http://myget.org/gallery/xunit

**Comment by Gillardo** [comment]
Thanks, i have now added this to my nuget.config file, but even though my project builds, none of my tests now appear in vs2015...

**Comment by Gillardo** [comment]
All i get is this under output
```
------ Load Playlist started ------
========== Load Playlist finished (0:00:00.0020147) ==========
------ Discover test started ------
========== Discover test finished: 0 found (0:00:01.9775114) ==========
```

**Comment by bradwilson** [comment]
Sounds like I'll need to make a new RC2 build, something has clearly changed...

**Comment by Gillardo** [comment]
Can you please let me know once this is done, that would be great!
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
This
email has been sent from a virus-free computer protected by Avast.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DDB4FAA8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by bradwilson** [comment]
Are you using `dnx` or `dotnet`? I just updated to DNX `1.0.0-rc2-16357` and everything continues to compile and unit tests continue to run.

**Comment by bradwilson** [comment]
I also don't see anything here that looks like it should be breaking us:

https://github.com/aspnet/announcements/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0-rc2

**Comment by Gillardo** [comment]
```
    "frameworks": {
        "dnx46": { }
    },
```

**Comment by Gillardo** [comment]
Here is one of my project.json files from my unit testing project

```
{
    "version": "1.0.0-*",
    "projectUrl": "",
    "licenseUrl": "",

    "dependencies": {
        "xunit": "2.0.0-rc2-*",
        "xunit.runner.dnx": "2.1.0-rc2-*",
        "Microsoft.Extensions.Logging": "1.0.0-rc2-*",
        "Microsoft.Extensions.Logging.Console": "1.0.0-rc2-*",
        "Microsoft.Extensions.Logging.Debug": "1.0.0-rc2-*",

        "my.Service.ProjectName": "1.0.0-*",

        "Moq": "4.2.1507.118"
    },

    "commands": {
        "test": "xunit.runner.dnx"
    },

    "frameworks": {
        "dnx46": {
            "dependencies": {
                "System.Collections": "4.0.11-rc2-*",
                "System.Linq": "4.0.1-rc2-*",
                "System.Threading": "4.0.11-rc2-*",
                "System.Runtime": "4.0.21-rc2-*",
                "Microsoft.CSharp": "4.0.1-rc2-*"
            }
        }
    }
}
```

**Comment by bradwilson** [comment]
This:

`"xunit": "2.0.0-rc2-*"`

Should be this:

`"xunit": "2.1.0"`

Otherwise, I don't see anything unexpected...

**Comment by margagn** [comment]
I am having the exact same issue. I am also trying to migrate to RC2.

```
Error discovering tests in 'D:\DataAccessLayer.Tests\project.json'. 
Exception: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: System.MissingMethodException: Method not found: 'Microsoft.Extensions.Logging.ILogger Microsoft.Extensions.Logging.LoggerFactoryExtensions.CreateLogger(Microsoft.Extensions.Logging.ILoggerFactory)'.
   at Microsoft.Dnx.TestHost.TestServices.CreateTestServices(Project project, ReportingChannel channel)
   at Microsoft.Dnx.TestHost.Program.<>c__DisplayClass0_0.<<Main>b__0>d.MoveNext()
   --- End of inner exception stack trace ---
``` 

My project.json looks like this:

```
  "dependencies": {
    "MRG.Minto.DataAccessLayer": "1.0.0-*",
    "MRG.Minto.Service": "1.0.0-*",

    "Newtonsoft.Json": "7.0.1",

    "xunit": "2.1.0-*",
    "xunit.runner.dnx": "2.1.0-*",

    "Microsoft.AspNet.TestHost": "1.0.0-rc2-16150",
    "Microsoft.AspNet.Hosting": "1.0.0-rc2-16150",
    "Microsoft.Dnx.Runtime": "1.0.0-rc2-16357",

    "Microsoft.Dnx.Runtime.Abstractions": "1.0.0-beta8"
  },
```

**Comment by Gillardo** [comment]
Changing dependencies to xunit 2.1.0 did fix the problem.  Thank you

**Comment by Gillardo** [comment]
Seem to have this problem again but i have now updated my frameworks to this
```
 "frameworks": {
    "net451": { }
  }
```

When i do `dotnet restore`  i get this output.  Any idea??
![test](https://cloud.githubusercontent.com/assets/7406381/13327862/9a5d65ce-dbe3-11e5-8017-59cb6994586a.gif)

**Comment by bradwilson** [comment]
We are not planning to issue any releases until after ASP.NET Core RC 2 has shipped. Sorry.

**Comment by Gillardo** [comment]
Any news on this yet?

**Comment by bradwilson** [comment]
Has ASP.NET Core RC 2 shipped yet? :)

**Comment by taspeotis** [comment]
I'm using `dnx-clr-win-x86.1.0.0-rc1-update1`. My `dependencies` is below:

```
  "dependencies": {
    "another-project-name-here": "1.0.0-*",
    "xunit": "2.1.0",
    "xunit.runner.dnx": "2.1.0-rc1-build204"
  },
```

As best I can tell, this is correct as per https://xunit.github.io/docs/getting-started-dnx.html and anything else is wrong.

> Note: Specific versions of xUnit.net only support specific versions of DNX

>     (DNX version) 1.0.0-rc1-update1
    (xunit packages) 2.1.0
    (xunit.runner.dnx package) 2.1.0-rc1-build204

And the output:

```
Exception: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: System.MissingMethodException: Method not found: 'Microsoft.Extensions.Logging.ILogger Microsoft.Extensions.Logging.LoggerFactoryExtensions.CreateLogger(Microsoft.Extensions.Logging.ILoggerFactory)'.
   at Microsoft.Dnx.TestHost.TestServices.CreateTestServices(Project project, ReportingChannel channel)
   at Microsoft.Dnx.TestHost.Program.<>c__DisplayClass0_0.<<Main>b__0>d.MoveNext()
   --- End of inner exception stack trace ---
```

**Comment by bradwilson** [comment]
Does following the instructions on the web site work for you? If so, then the trick is tracking down the delta between what you're doing and what the sample is doing.

That said, DNX is dead. We won't be fixing any bugs, if you happen to find one, because everything moves to the .NET CLI in RC 2. We will re-evaluate any issues you might find with DNX in the context of the new client, once RC 2 ships (and once we have a new runner).

**Comment by JvanderStad** [comment]
rc2 has shipped ;)

**Comment by bradwilson** [comment]
And so has our support.

https://xunit.github.io/docs/getting-started-dotnet-core.html

**Comment by Gillardo** [comment]
Having problems updating my tests.  My projects that i want to test have the following settings in project.json
```
"frameworks": {
    "net451": { }
}
```

But on the getting started page, if says that your project must be setup using `netcoreapp1.0`, but my project isnt compatible with this project type.

**Comment by Gillardo** [comment]
How can i get my xunit test projects to work with a project using framework `net451`? (Sorry for closing as pressed wrong button)

**Comment by Gillardo** [comment]
Any news on this please?

**Comment by jaredcnance** [comment]
I am having the same issues. My projects are dependent upon libraries that are not yet compatible with the netcoreapp framework. When I build my project I get the following stack trace during test discovery:
```
------ Discover test started ------
Discovering tests in 'C:\Code\...\project.json' ["C:\Program Files (x86)\dotnet\dotnet.exe" test "C:\Code\...\project.json" --output "C:\Code\...\bin\Debug\net451" --port 51066 --parentProcessId 5932 --no-build]
Unable to start C:\Program Files (x86)\dotnet\dotnet.exe
dotnet-test Error: 0 : [ReportingChannel]: Error sending System.IO.IOException: Unable to write data to the transport connection: Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.. ---> System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.
   at System.Net.Sockets.Socket.Send(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags, SocketError& errorCode)
   at System.Net.Sockets.Socket.Send(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   --- End of inner exception stack trace ---
   at System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   at System.IO.Stream.WriteByte(Byte value)
   at System.IO.BinaryWriter.Write(Byte value)
   at System.IO.BinaryWriter.Write7BitEncodedInt(Int32 value)
   at System.IO.BinaryWriter.Write(String value)
   at Microsoft.DotNet.Tools.Test.ReportingChannel.Send(Message message)
dotnet-test Error: 0 : [ReportingChannel]: Waiting for message failed System.IO.IOException: Unable to read data from the transport connection: An established connection was aborted by the software in your host machine. ---> System.Net.Sockets.SocketException: An established connection was aborted by the software in your host machine
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   --- End of inner exception stack trace ---
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.IO.Stream.ReadByte()
   at System.IO.BinaryReader.ReadByte()
   at System.IO.BinaryReader.Read7BitEncodedInt()
   at System.IO.BinaryReader.ReadString()
   at Microsoft.DotNet.Tools.Test.ReportingChannel.ReadMessages()
```

**Comment by bradwilson** [comment]
Use a CI build (see #843) and see if that fixes it.

**Comment by bradwilson** [comment]
It appears that Visual Studio integration is broken if you have two frameworks specified in your `project.json` for your unit test. If you specify just a single framework, everything works as expected. Command line tests with multiple frameworks works as expected.

That means this is a bug with the .NET tooling, and should be reported in their repo.

**Comment by Gillardo** [comment]
Seem to have a different error now.  For some reason my project would build as it said it needed a reference the `System.Runtime`, so i have added that and it builds.  Now when i try to run `dotnet test` i get an error message saying

> E:\Github\My.Project.Tests>dotnet test
> Project My.Project.Tests (.NETFramework,Version=v4.5.1) was previously compiled. Skipping compilation.
> xUnit.net .NET CLI test runner (64-bit Desktop .NET win10-x64)
>   Discovering: my.Project.Tests
> System.InvalidOperationException: Unknown test framework: could not find xunit.dll (v1) or xunit.execution.*.dll (v2) in E:\Github\My.Project.Te
> sts\bin\Debug\net451

My project.json looks like this for my test project

```
{
    "name": "My.Project.Tests",
    "version": "1.0.0-alpha-1",
    "testRunner": "xunit",

    "dependencies": {
        "xunit": "2.1.0",
        "dotnet-test-xunit": "1.0.0-rc3-*",
        "My.Project": "1.0.0-*",
        "Moq": "4.5.0"
    },

    "runtimes": {
        "win8-x64": { }
    },

    "frameworks": {
        "net451": {
            "dependencies": {
                "Microsoft.NETCore.Platforms": "1.0.1-rc2-24027"
            },
            "frameworkAssemblies": {
                "System.Runtime": { "type": "build" }
            }
        }
    },

    "buildOptions": {
        "copyToOutput": {
            "include": [ "xunit.runner.json" ]
        }
    }
}
```

Is there something i have missed?  This all worked fine when i used to use `dnx451` but since moving to `net451` and the dotnet cli, it has gone to plonk!

Any help would be appreciated, as all my test projects have been on hold now for weeks

**Comment by jaredcnance** [comment]
Upgrading to the latest build worked for me. I can now run tests in VS. Thanks.

**Comment by Gillardo** [comment]
Whats the latest build?  I guess you just mean to `dotnet restore`?

**Comment by jaredcnance** [comment]
http://myget.org/feed/xunit/package/nuget/dotnet-test-xunit
You may need to clear the NuGet cache before running dotnet restore.

**Comment by Gillardo** [comment]
still doesnt work for me. may try recreating my project and importing code, maybe a funny somewhere
